src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlTokenValidationParameters.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using CoreWCF.IdentityModel.Selectors;
+using Microsoft.IdentityModel.Tokens;
+using MsIdentityTokens = Microsoft.IdentityModel.Tokens;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    internal class SamlTokenValidationParameters
+    {
+        private const string CERTIFICATE_VALIDATOR = "Certificatevalidator";
+        private const string SAML_SECURITY_TOKEN = "SamlSecurityToken";
+        private const string ISSUER_NAME_REGISTRY = "IssuerNameRegistry";
+
+        public SamlTokenValidationParameters()
+        {
+        }
+
+        internal MsIdentityTokens.TokenValidationParameters ConvertToTokenValidationParameters
+            (SecurityTokenHandlerConfiguration securityHandlerConfiguration, SecurityToken securityToken)
+        {
+            MsIdentityTokens.TokenValidationParameters tokenValidationParams = new MsIdentityTokens.TokenValidationParameters();
+
+            tokenValidationParams.ValidateAudience = (securityHandlerConfiguration.AudienceRestriction?.AudienceMode) != AudienceUriMode.Never;
+            if (tokenValidationParams.ValidateAudience)
```

Comment:
This logic is different than is in SamlSecurityTokenRequirement.ShouldEnforceAudienceRestriction. Does that have any effect? Are we going to either pass the check when we shouldn't, or fail the check when we shouldn't be checking?